### PR TITLE
Kill browser bundle

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

Removes the tiny browser bundle that bootstraps each page. Instead each page will have its own entry point.

For now I have achieved this by creating a Webpack config for each page, and using the `string-replace-loader` Webpack plugin to inject the path to each page from the `browser.js` bootstrap module.

## Why?

Clarifies the loading strategy

## Future ideas

- [x] Rather than hardcode the list of page names in the Webpack config, this could be generated dynamically

### CODE REVIEW

File: __config__/webpack/webpack.config.js [hunk 1]
```diff
@@ -7,57 +7,76 @@ const { smart: merge } = require('webpack-merge');
...
-        app: [
-            'preact-emotion', // force preact-emotion into the vendor chunk
-            `./src/${platform}.js`,
+const config = ({ platform, pages }) => {
+    const entry =
+        platform === 'server'
+            ? {
+                  app: [
+                      'preact-emotion', // force preact-emotion into the vendor chunk
+                      `./src/${platform}.js`,
```
Comment (GHaberis) [inline __config__/webpack/webpack.config.js]: this could just be `./src/server.js` now?

Comment (SiAdcock) [reply __config__/webpack/webpack.config.js]: Yeah, could do. This is me being too DRY!

File: src/browser.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ import { render } from 'preact';
 import { hydrate as hydrateCSS } from 'emotion';
 import createStore from 'unistore';
 import { Provider } from 'unistore/preact';
+// eslint-disable-next-line import/no-extraneous-dependencies,import/no-unresolved,import/extensions
+import Page from '__PAGE__ENTRY__POINT__';
```
Comment (sndrs) [inline src/browser.js]: maybe a little comment in here explaining and pointing to webpack config?

File: __config__/webpack/webpack.config.js [hunk 3]
```diff
@@ -67,4 +89,13 @@ module.exports = [
         },
         envConfig.server,
     ),
+    ...pages.map(page =>
```
Comment (sndrs) [inline __config__/webpack/webpack.config.js]: this means we'll have a config per page i think? do we just want mtuliple entry points in one config? i think that will bundle quicker too

Comment (SiAdcock) [reply __config__/webpack/webpack.config.js]: Yeah, that was my first attempt. I hit a blocker:
```js
{
  entry: {
    'Article': './src/browser.js',
    'Article.immersive': './src/browser.js',
  },
  // ...
  {
    module: {
      rules: [
        {
           test: /browser\.(js|jsx)$/,
           loader: 'string-replace-loader',
           options: {
             search: '__PAGE__ENTRY__POINT__',
             replace: `pages/Article.js`, // <-- how do we target the right entry point?
           },
        },
      ]
    }
  }
}

```

Comment (sndrs) [reply __config__/webpack/webpack.config.js]: would this work?
```js
entry: pages.reduce(
    (entries, page) => ({
        ...entries,
        [page]: `string-replace-loader?search=__PAGE__ENTRY__POINT__&replace=pages/${page}!./src/browser.js`,
    }),
    {},
),
```

Comment (sndrs) [reply __config__/webpack/webpack.config.js]: yeah looks like it!

Comment (sndrs) [reply __config__/webpack/webpack.config.js]: https://github.com/guardian/guui/compare/sa-browser-side-effects...inject-page

Comment (sndrs) [reply __config__/webpack/webpack.config.js]: that might have too much moving stuff about in it so do kill that branch if it's confusing. but the loader seems to work though

**Comment by sndrs** [comment]
@SiAdcock have merged the housekeeping changes in #50 to master and fixed the conflicts in your branch – you're up to date with master now

**Comment by SiAdcock** [comment]
This is awesome, thanks @sndrs! So much tidier

File: __config__/webpack/webpack.config.js (hunk 3 above)
Comment (SiAdcock) [reply __config__/webpack/webpack.config.js]: No, it's great, the `helpers` tidy things up a lot! I've merged your changes into this PR.

**Pull request merged**